### DISCUSSION TITLE
Fungal -RiPPs- Like

### DISCUSSION THREAD

**Discussion by sawsanabed** [medema-group/BiG-SCAPE#113] [Q&A]
Hello all,
I am using the antiSMASH V.7 and in this version, there is a new group of RiPPs under the name (Fungal-RiPPs- like). the issue is when I run the big-scape it puts these BGCs into the " other" groups and it does not show it as a RiPPs on the HTML link. Is there any way to overcome this issue 
![Fungal-ripps-like](https://github.com/medema-group/BiG-SCAPE/assets/150505296/2cc3d362-3d9d-4117-ab8b-a826c088952c)
Thanks a lot in advance

**Comment by jorgecnavarrom** [comment]
Hi
With the upcoming version 2 of BiG-SCAPE, we haven't updated the new detection types from antiSMASH 7. You could manually add this type (`fungal-RiPP-like`) to the list of RiPPs in the `sort_bgc` function of the `functions.py` file (currently [here, at line 391](https://github.com/medema-group/BiG-SCAPE/blob/master/functions.py#L391), but line number might be different depending on the version you're using):
```python
def sort_bgc(product):
    """Sort BGC by its type. Uses antiSMASH annotations
    (see 
    https://docs.antismash.secondarymetabolites.org/glossary/#cluster-types)
    """
  ...
    ripps_products = {'lantipeptide', 'thiopeptide', 'bacteriocin', 'linaridin', 
                     'cyanobactin', 'glycocin', 'LAP', 'lassopeptide', 
                     'sactipeptide', 'bottromycin', 'head_to_tail', 'microcin', 
                     'microviridin', 'proteusin', 'lanthipeptide', 'lipolanthine', 
                     'RaS-RiPP', 'fungal-RiPP', 'TfuA-related', 'guanidinotides', 
                     'RiPP-like', 'lanthipeptide-class-i', 'lanthipeptide-class-ii', 
                     'lanthipeptide-class-iii', 'lanthipeptide-class-iv',
                     'lanthipeptide-class-v', 'ranthipeptide', 'redox-cofactor',
                     'thioamitides', 'epipeptide', 'cyclic-lactone-autoinducer',
                     'spliceotide', 'RRE-containing', 'crocagin', 'fungal-RiPP-like'}
    saccharide_products = {'amglyccycl', 'oligosaccharide', 'cf_saccharide', 
                     'saccharide'}
...
```

**Comment by lconreur** [comment]
Hi,
Different person, same "issue". :) (using Bigscape-1.1.9 I can add 'fungal-RiPP-like' into ripps_products. I guess other categories below can go to 'Others'?

Warning: unknown product 'terpene-precursor'
Warning: unknown product 'NRP-metallophore'
Warning: unknown product 'isocyanide'
Warning: unknown product 'isocyanide-nrp'
Warning: unknown product 'RCDPS'

thank you!

**Reply by nlouwen** [reply]
Hi!
That would indeed be an option, it mostly depends on which comparisons you would like to make (if you ran with `classify`). For example, if you would want to compare all NRPS BGCs to the NRP-metallophore, the NRPS category might be better.
Alternatively, you could try BiG-SCAPE version 2, where you can bin directly based on antiSMASH classes or categories (see [wiki](https://github.com/medema-group/BiG-SCAPE/wiki/06.-BiG-SCAPE-Classes#classify-based-on-antismash-classcategory)).

**Comment by lconreur** [comment]
Thank you, I will go for BiG-SCAPE version 2 :-)

**Comment by lconreur** [comment]
Hello again (6 months later) !
I'm back on this topic. I am wondering if we could kind of "map" classes or categories to defined bin_labels. 
i.e. if I have "other.terpene" as an antismash category and I would like it to be under bin_label "terpene" ? Kind of "personalized bin_label".

My idea behind is to empty bin_label "other". When I look deeper into BGCs, I see I could reclass most of them.

Thank you :)
(I use BiG-SCaPE v2.0.0-beta7)

**Reply by nlouwen** [reply]
Hi!
Such a custom mapping is currently not possible, partly because these labels are based on antiSMASH defined classes/categories. If you have suggestions to reclassify "other" BGCs, then that should probably be done at the antiSMASH level, i.e. resulting in gbk files with your desired labels. I believe you can run antiSMASH with custom rules, so you could simply change the existing "other" rules to any label you'd want. (Keep in mind the reproducibility issues here)
Apart from that, your example to have "other.terpene" show up in the terpene bin could be achieved using `--hybrids-off`. When using this flag, any hybrid BGC, e.g. "other.terpene", will be added to both "other" and "terpene", instead of the hybrid "other.terpene" bin.

**Reply by lconreur** [reply]
Hi Nico,
again, thanks a lot for your answer. I'm still trying to understand the subtleties of both tools (antiSMASH and BIGSCaPE), that's why I don't have more global questions yet. I REALLY appreciate your help here.
I'll try with `--hybrids-off`.
Have a nice day and a nice year!
Laurence
